### ISSUE TITLE
Electron calo hits duplicated as gammas

### ISSUE THREAD

**Issue by cherylepatrick** [SuperNEMO-DBD/Falaise#213]
Apologies if this is a duplicate - it was actually @wausain that found it but I don't think it's already been logged. Hamzah found a few events where the same calorimeter hits had been reconstructed as both and electron and a gamma hit (with the same time and energy). The gamma must be a bug. Here's a picture from a talk of Hamzah's:
![image](https://user-images.githubusercontent.com/18677866/86244883-880df900-bba0-11ea-8d08-2fceb38255c4.png)

Particles 0 (negative/electron) and 3 (neutral/gamma) are the same calorimeter hit. Also notice: there are particles 0, 1 and 3 but not particle 2. Is that odd? Looks odd to me.

**Comment by drbenmorgan** [comment]
Seems reconstruction and likely GammaTracking/Clustering related, so assigning to @pfranchini and @Tedjditi for triage and fix. Would help them if you can supply - exact Falaise setup used, file with 5-10 events to replicate the bug (if triaged as one)

**Comment by cherylepatrick** [comment]
Like I say, I haven't seen this myself, we need input from @wausain !

**Comment by drbenmorgan** [comment]
Assigned @wausain as well!

**Comment by wausain** [comment]
Hi All

I used Falaise 4.0 to simulate and reconstruct these events.

These events are very infrequent, I have a total of 81 events from 10 Billion simulations. I have these files in my hpss/sps space on cclyon if needed. How do you want to go about reproducing them?

The simulation configuration I used was:

_#@format=datatools::configuration::variant
#@format.version=1.0
#@organization=snemo
#@application=falaise


[name="flsimulate" type="flsimulate::section"]
numberOfEvents : integer = 1000000



[name="flsimulate.variantService" type="flsimulate::section"]

settings : string[3] = \
"geometry:layout/if_basic/magnetic_field=false" \
"vertexes:generator=pmt_8inch_main_wall_glass_bulk" \
"primary_events:generator=Tl208"_

**Comment by pfranchini** [comment]
I am a bit discouraged by having to reproduce 120M events before getting one bad event ;)

**Comment by cherylepatrick** [comment]
Yeah I think that ain't gonna happen. @wausain can you point us at the existing files and let us know the event numbers of the naughty ones?

**Comment by wausain** [comment]
Hi 

All events can be found as tarred files at: 

**/hpss/in2p3.fr/group/nemo/users/hhussain/8inch_PMT_Externals_10B/Glass_Bulk/Tl208/No_Field/** using the rfdir command

I have already extracted a few event in the corresponding sps directory:
**/sps/nemo/scratch/hhussain/8inch_PMT_Externals_10B/Glass_Bulk/Tl208/No_Field**

There are a total of 81 events from which there are 12 events which display this potential issue. 

They can be found at /sps/nemo/scratch/hhussain/8inch_PMT_Externals_10B/Glass_Bulk/Tl208/No_Field : 

**run_852**/Tl208_8inch_pmt_glass_bulk_0g_rec_852.brio -> event number 974344

**run_1105**/Tl208_8inch_pmt_glass_bulk_0g_rec_1105.brio -> event number 213245

**run_4250**/Tl208_8inch_pmt_glass_bulk_0g_rec_4250.brio -> event number 681865

**run_6487**/Tl208_8inch_pmt_glass_bulk_0g_rec_6487.brio -> event number 869449

**run_6893**/Tl208_8inch_pmt_glass_bulk_0g_rec_6893.brio -> event number 567437

**run_6920**/Tl208_8inch_pmt_glass_bulk_0g_rec_6920.brio -> event number 376641

**run_7277**/Tl208_8inch_pmt_glass_bulk_0g_rec_7277.brio -> event number 103001

**run_7406**/Tl208_8inch_pmt_glass_bulk_0g_rec_7406.brio -> event number 579297

**run_9149**/Tl208_8inch_pmt_glass_bulk_0g_rec_9149.brio -> event number 806629

**run_9257**/Tl208_8inch_pmt_glass_bulk_0g_rec_9257.brio -> event number 339590

currently missing two of the events!

Sorry for the long post but if you have any issues accessing these events, feel free to contact me.

**Comment by cherylepatrick** [comment]
AMAZING, thank you @wausain !

**Comment by pfranchini** [comment]
Hi @Tedjditi I was trying to look at this but I am wandering in the dark. Do you have any reference where I should start from? Thanks
